java/src/jmri/jmrit/logix/Engineer.java

```diff
@@ -1011,43 +1010,40 @@ private void checkerDone(Warrant oldWarrant, Warrant newWarrant) {
             log.info(msg);
         }
         Engineer.setFrameStatusText(msg, color, true);
-        try {
-            _checker.join(100);
-        } catch (InterruptedException ie) {
-        } finally {
-            _checker = null;
-        }
+        _checker = null;
     }
 
     private class CheckForTermination extends Thread {
         Warrant oldWarrant;
         Warrant newWarrant;
-        long timeLimit;
+        long waitTime; // time to finish remaining commands
 
         CheckForTermination(Warrant oldWar, Warrant newWar, int num, long limit) {
             oldWarrant = oldWar;
             newWarrant = newWar;
-            timeLimit = limit;
-            if (log.isDebugEnabled()) log.debug("checkForTermination({}, {}, {}, {})",
-                    oldWarrant.getDisplayName(), newWarrant.getDisplayName(), num, timeLimit);
+            waitTime = limit;
+            if (log.isDebugEnabled()) log.debug("checkForTermination of {}, waitTime= {})",
+                    oldWarrant.getDisplayName(), newWarrant.getDisplayName(), waitTime);
```

Comment:
Static analysis fails on this line. The call to `log.debug()` has two `{}` but three parameters besides the format string. It seems that you have one `{}` too little.